gridstatus/aeso/aeso.py

```diff
@@ -242,3 +243,140 @@ def get_asset_list(
+        end: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp] | None = None,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        """
+        Get pool price data.
+
+        Returns:
+            DataFrame containing pool price data
+        """
+        if date == "latest":
+            return self.get_pool_price(date="today")
+        return self._get_pool_price_data(date, end, actual_or_forecast="actual")
+
+    @support_date_range(frequency=None)
+    def get_forecast_pool_price(
+        self,
+        date: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp],
+        end: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp] | None = None,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        """
+        Get pool price data.
+
+        Returns:
+            DataFrame containing pool price data
+        """
+        if date == "latest":
+            return self.get_forecast_pool_price(date="today")
+        return self._get_pool_price_data(date, end, actual_or_forecast="forecast")
+
+    def _get_pool_price_data(
+        self,
+        start_date: str,
+        end_date: str | None = None,
+        actual_or_forecast: Literal["actual", "forecast"] = "actual",
+    ) -> pd.DataFrame:
+        """
+        Get pool price data.
+
+        Returns:
+            DataFrame containing pool price data
+        """
+        start_date = pd.Timestamp(start_date).strftime("%Y-%m-%d")
+        end_date = pd.Timestamp(end_date).strftime("%Y-%m-%d") if end_date else None
+
+        endpoint = f"poolprice-api/v1.1/price/poolPrice?startDate={start_date}"
+        if end_date:
+            endpoint += f"&endDate={end_date}"
+        data = self._make_request(endpoint)
+        df = pd.json_normalize(data["return"]["Pool Price Report"])
+        df["Interval Start"] = pd.to_datetime(
+            df["begin_datetime_utc"],
+            utc=True,
+        ).dt.tz_convert(self.default_timezone)
+        df["Interval End"] = df["Interval Start"] + pd.Timedelta(hours=1)
+        df = df.rename(
+            columns={
+                "pool_price": "Pool Price",
+                "forecast_pool_price": "Forecast Pool Price",
+            },
+        )
+
+        if actual_or_forecast == "actual":
+            df["Pool Price"] = pd.to_numeric(df["Pool Price"], errors="coerce")
+            df = df[df["Pool Price"].notna()]
+            return df[["Interval Start", "Interval End", "Pool Price"]]
+        else:
+            df["Forecast Pool Price"] = pd.to_numeric(
+                df["Forecast Pool Price"],
+                errors="coerce",
+            )
+            # NB: Publish times are a bit opaque from AESO, so we calculate a best estimate here.
+            # Forecast pool price is provided for current and next two hours, updated every 5 minutes.
+            # For future intervals: use request time floored to 5 minutes
+            # For past/current intervals: use 5 minutes before interval start
```

Comment:
Just wondering, where did you find out how often the forecast is updated? I didn't see it on the API page.

https://developer-apim.aeso.ca/api-details#api=poolprice-api-v1-1&operation=getPoolPriceDateRangeReport

Reply:
Sorry, should have read the PR description.

http://ets.aeso.ca/Market/Reports/Manual/HelpText/current-actual-forecast-metadata.pdf